### ISSUE TITLE
flux:form icon-Resource Path mismatch in Page-View <> Content-Element Dropdown

### ISSUE THREAD

**Issue by typoworx-de** [FluidTYPO3/flux#891]
I noticed a weird path issue for using an icon in flux:form:

```
<v:variable.set name="typeIcon" value="{f:uri.resource(extensionName: 'my_extkey', path: 'Icons/bootstrap/glyphicons_113_justify.png')}" />
<flux:form id="accordion" options="{useTypoScript: 0, group: 'Bootstrap', icon: '{typeIcon}'}">
 ..Some Form Stuff...
</flux:form>

<f:section name="Preview">
	<v:variable.set name="typeIcon" value="{f:uri.resource(extensionName: 'my_extkey', path: 'Icons/bootstrap/glyphicons_113_justify.png')}" />
	<f:image src="{typeIcon}" />
	<flux:widget.grid />
</f:section>
```
*Note:
For better debugging I used v:variable instead of f.uri.resource directly. both tried with same result!*

**I would expect that f.uri.resource ALWAYS generates a correct path in any scope...**
but the result ist as follows:

**In TYPO3 Page Module, the Type-Icon is missing as the path is incorrect there:**
img src refers to "../typo3conf/ext/my_extkey/Resources/Public/Icons/bootstrap/glyphicons_113_justify.png".

**But the url will resolve as:**
/typo3/sysext/cms/typo3conf/ext/my_extkey/Resources/Public/Icons/bootstrap/glyphicons_113_justify.png

which is definitely wrong!

I did the same in the preview-tag of flux with an f:image... and surprise.. there it's working!
/typo3conf/ext/my_extkey/Resources/Public/Icons/bootstrap/glyphicons_113_justify.png

I used the same syntax and also used v:variable in both cases (with and without) tested...

**In Page-Wizard** the icon has another path-bug at is definitly generating an url containing the whole absolute server path in it!!!
../../..//srv/www/clients/client1/web156/web/typo3conf/ext/my_extkey/Resources/Public/Icons/bootstrap/glyphicons_155_show_big_thumbnails.png

**Comment by glucka** [comment]
i have the same problem

**Comment by rabe69** [comment]
me too

**Comment by julianxhokaxhiu** [comment]
+1

**Comment by NamelessCoder** [comment]
Thanks for the detailed report! A few things:

> I would expect that f.uri.resource ALWAYS generates a correct path in any scope...

Although that is a reasonable expectation, in real life it doesn't work that way. Whatever the reason is, `f:uri.resource` will indeed generate incorrect URLs when used from certain places within the TYPO3 backend. The cause being the *location of the script that generates the image tag*. If this script resides in a location that is not exactly the same as in the frontend, the URI will be incorrect.

This is, however, not a FluidTYPO3 issue but a TYPO3 core issue. It affects any use of Fluid templates with `f:uri.resource` and others, if that template is rendered in (just for example) the page module.

Furthermore: the recent change in TYPO3 to remove all alternative ingress points will have solved this one in future versions of TYPO3. But, to get this one solved in 6.2 you would have to justify to the core team that it is in fact a bug. They might argue that `f:uri.resource` was never intended for backend use, but I would personally vote for getting a fix for this one.

> In TYPO3 Page Module, the Type-Icon is missing as the path is incorrect there:

This one is solved in the current Flux development branch and will be released to TER as part of the next (bugfix, 6.2-compatible) release. It is caused by the icon utility not being available on older Flux versions.

> In Page-Wizard the icon has another path-bug at is definitly generating an url containing the whole absolute server path in it!!!

Also solved in Flux development branch and headed for TER soon.

Will close the issue - problems are either resolved and soon to be released, or are problems in the TYPO3 core itself over which we don't have control.

**Comment by typoworx-de** [comment]
This bug seems to be still present. I was unable to get a working as expected in TYPO3 v.8 and flux 8.2.1, fluidpages 4.1.0 and fluidcontent 5.2.0.

Is there any new for this?
